packages/big-design/src/components/MultiSelect/spec.tsx

```diff
@@ -369,6 +373,29 @@ test('clicking on disabled select options should not trigger onItemClick', async
   await waitForElement(() => screen.getByRole('option', { name: /mex/i }));
 });
 
+test('opening the MultiSelect triggers onOpen', async () => {
+  const { getAllByRole, queryByRole } = render(MultiSelectMock);
+  const button = getAllByRole('button')[2];
+
+  fireEvent.click(button);
+
+  expect(onOpen).toHaveBeenCalled();
+
+  await waitForElement(() => queryByRole('listbox'));
+});
+
+test('closing the MultiSelect triggers onClose', async () => {
+  const { getAllByRole, queryByRole } = render(MultiSelectMock);
+  const button = getAllByRole('button')[2];
+
+  fireEvent.click(button);
+  fireEvent.click(button);
+
+  expect(onClose).toHaveBeenCalled();
+
+  await waitForElement(() => queryByRole('listbox'));
```

Comment:
`queryBy*` should only be used for querying when the element is not present: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence


in this case we should use `findByRole` which does the wait for you:

```tsx
const listbox = await findByRole('listbox');

expect(listbox).toBeInTheDocument();
```